### DISCUSSION TITLE
How can we  verifying the liveness of a device?

### DISCUSSION THREAD

**Discussion by maunishpatel1** [openthread/openthread#8675] [Q&A]
Hello @jwhui,

We are connecting **20+ end devices** in an open-thread network. For verifying the liveness of devices we added the mechanism in which the devices will send the periodical message**(PING messages)** every **1-minute** to the server. If the server doesn't receive the message from the device in a specific time(2 minutes), the device will be declared offline. 

This mechanism works fine for fewer devices like **5 devices** but after increasing the number of devices to 20 the traffic increases due to the ping messages and some of the **end devices get disconnected** from the open thread network. Also, the network's functionality is getting affected.

**Is there any better way to check the device status without burdening the network RF?**

Please provide your quick suggestion or solution as we are in the middle of the deployment.

**Thanks In Advance,**

**Comment by jwhui** [comment]
For a simple topology, a Thread network should support having 20 devices reporting every minute.

Can you provide more detail on your setup?
1. Git commit id
2. IEEE 802.15.4 hardware platform
3. Build steps
4. Network topology
5. Packet capture for analysis in Wireshark

**Comment by maunishpatel1** [comment]
Thanks for the response,

**For a simple topology, a Thread network should support having 20 devices reporting every minute**.

* Is there any API for accessing or observing the reporting?
* What if we are using more than 20 devices like 100 devices?

**Can you provide more detail on your setup?**

* We are forming the open thread network in RCP mode. We are using the custom board as a Host and the k32w061 module as RCP. As end devices, we are using k32w061.

**1: Git Commit id:-** 
		* For otbr-agent we are using **83babaf236cad8471be28185d8d4351d37564919** commit.
		* For RCP using **ced158e65a00dd5394c04548b7b187d3a3f11ee**

**NOTE:- These are quite older commits but it was working fine for us. But after adding the ping mechanism it is not working as expected.**

**2: IEEE 802.15.4 hardware platform:-**
		* Using the custom board as Host and [k32w061](https://www.nxp.com/products/wireless/multiprotocol-mcus/k32w061-41-high-performance-secure-and-ultra-low-power-mcu-for-zigbeethread-and-bluetooth-le-5-0-with-built-in-nfc-option:K32W061_41) as RCP.  
		* Using k32w061 devices as end devices.

**3: Build steps:-**
		* Using the below steps for building the RCP and end device binary.
		https://github.com/openthread/ot-nxp/blob/main/src/k32w0/k32w061/README.md

**4: Network topology:-**
		* We are forming an open thread network with more than 20 **REED devices**. Max router count **configured to 6**.

**5:Packet capture for analysis in Wireshark:-**
Please refer to the below captured Wireshark packets.
[wireshark_logs.zip](https://github.com/openthread/openthread/files/10478043/wireshark_logs.zip)

**Reply by jwhui** [reply]
> Is there any API for accessing or observing the reporting?

There are a number of CLIs that can provide various information about neighboring routers, children, etc. Review the various [CLI READMEs](https://github.com/openthread/openthread/tree/main/src/cli) for more information.

> What if we are using more than 20 devices like 100 devices?

It could also support 100 devices, but largely depends on the amount of data being sent. 100 devices sending once per minute is approaching 2 messages per second.

> Please refer to the below captured Wireshark packets.

Without the network key, it was difficult to decode many packets.

That said, there is **a lot** of IEEE 802.15.4 Beacon traffic in your packet capture. I would suggest eliminating devices sending Beacon Request messages to avoid consuming lots of channel capacity on Beacon messages.
